Lib/test/test_weakref.py

```diff
@@ -116,6 +116,33 @@ def test_basic_ref(self):
         del o
         repr(wr)
 
+    @support.cpython_only
+    def test_ref_repr(self):
+        obj = C()
+        ref = weakref.ref(obj)
+        self.assertEqual(repr(ref),
+                         f"<weakref at {id(ref):#x}; "
+                         f"to '{C.__module__}.{C.__qualname__}' "
+                         f"at {id(obj):#x}>")
```

Comment:
There's a failure on windows due to string representation of pointers.
Can we rewrite this like that?
```suggestion
        self.assertRegex(repr(ref),
                         rf"<weakref at 0x.* "
                         rf"to '{C.__module__}.{C.__qualname__}' "
                         rf"at 0x.*>")
```

Reply:
Right, I applied a similar fix.